src/main/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheck.java

```diff
@@ -80,7 +79,7 @@ public void visitToken(DetailAST ast) {
             final boolean extendsJlo =
                 ast.findFirstToken(TokenTypes.EXTENDS_CLAUSE) == null;
 
-            final boolean isUtilClass = extendsJlo && hasMethodOrField
+            final boolean isUtilClass = extendsJlo
                 && !hasNonStaticMethodOrField && hasNonPrivateStaticMethodOrField;
```

Comment:
IMO, this removal is valid.
`isUtilClass` has to be true for a violation, and  all 4 conditions have to be true for `isUtilClass` to be true.

If `hasMethodOrField` is false, the other 2 `has...MethodOrField` will also be false (because there is no method/field) and never make `isUtilClass` true.
If `hasMethodOrField` is true, the other 2  `has...MethodOrField` can be either true or false (depending on if there are private/static methods/fields).

So a violation is always dependent upon the other `has...MethodOrField` variables and this is why the result of `hasMethodOrField` doesn't matter and can be removed.

Reply:
ok, if we do not know a case, life will show us... in other case you are right.